lib/puppet/parser/functions/noop.rb

```diff
@@ -1,16 +1,16 @@
-# Set noop to true as default for current and children scopes
-Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default to true for all resources
+# Set noop to true as default for current and children scopes, or 'reset' an inherited default with noop(false)
+Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default for all resources
   in local scope and children scopes. This can be overriden in
   child scopes, or explicitly on each resource.
   ") do |args|
-
+      @noop_value = true if (@noop_value = args[0]).nil? or (![true, false].include? @noop_value)
```

Comment:
I'm not referring to string values, I see that they will result in the previous behavior. It appears as though when either boolean true or boolean false are passed, it's the same outcome. Or am I missing something.

Either way, could you please add a test for values other than boolean false, just to prove and verify the behavior. Also, I'd like to have it directly stated in the docs that any value other than boolean false results in original behavior. However I can add that later if necessary.